### PR TITLE
Bulk v2

### PR DESCRIPTION
**Pull request by g8rswimmer**
This is the implementation of the bulk 2.0 apis.

#30

### CODE REVIEW

File: bulk/bulk.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"errors"
+
+	"github.com/g8rswimmer/go-sfdc/session"
+)
+
+const bulk2Endpoint = "/jobs/ingest"
+
+// Resource is the structure that can be just to create bulk 2.0 jobs.
```
Comment (Guitarbum722) [inline bulk/bulk.go]: This comment is not very intuitive in my opinion.  Feel free to tell me to pound sand.

File: bulk/bulk.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+type Resource struct {
+	session session.ServiceFormatter
+}
+
+// NewResource create a new bulk 2.0 resource.  If the session is nil
```
Comment (Guitarbum722) [inline bulk/bulk.go]: "NewResource *creates..."

File: bulk/bulk.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	if session == nil {
+		return nil, errors.New("bulk: session can not be nil")
+	}
+	return &Resource{
+		session: session,
+	}, nil
+}
+
+// CreateJob will create a new bulk 2.0 job from the options that where passed.
+// The Job structure that is returned can be used to upload object data to the Salesforce org.
```
Comment (Guitarbum722) [inline bulk/bulk.go]: I would just say a "Job pointer" is returned (or something of the sort).  No need to say it's a "Job Structure"

File: bulk/formatter.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	if len(fields) == 0 {
+		return nil, errors.New("bulk formatter: fields are required")
+	}
+
+	f := &Formatter{
+		job:    job,
+		fields: fields,
+		sb:     strings.Builder{},
+	}
+	_, err := f.sb.WriteString(strings.Join(fields, job.delimiter()))
```
Comment (Guitarbum722) [inline bulk/formatter.go]: I _think_ this line could get combined with the `if` check if you prefer.

File: bulk/jobs.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+	NextRecordsURL string     `json:"nextRecordsUrl"`
+}
+
+// Jobs presents the response from the all jobs request.
+type Jobs struct {
+	session  session.ServiceFormatter
+	response jobResponse
+}
+
+func newJobs(session session.ServiceFormatter, parameters Parameters) (*Jobs, error) {
```
Comment (Guitarbum722) [inline bulk/jobs.go]: where is this used?

File: bulk/jobs.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+	sfdc "github.com/g8rswimmer/go-sfdc"
+	"github.com/g8rswimmer/go-sfdc/session"
+)
+
+// Parameters to query the bulk jobs.
+//
+// IsPkChunkingEnabled will filter jobs with PK chunking enabled.
+//
+// JobType will filter jobs based on job type.
```
Comment (Guitarbum722) [inline bulk/jobs.go]: Where does the filtering occur?

File: bulk/bulk.go (hunk 1 above)
Comment (g8rswimmer) [reply bulk/bulk.go]: I really don't know what to say in the description.  It is mainly a resource for creating bulk 2.0 jobs.

File: bulk/bulk.go (hunk 2 above)
Comment (g8rswimmer) [reply bulk/bulk.go]: 👍

File: bulk/bulk.go (hunk 3 above)
Comment (g8rswimmer) [reply bulk/bulk.go]: 👍

File: bulk/formatter.go (hunk 4 above)
Comment (g8rswimmer) [reply bulk/formatter.go]: thanks for the idea

File: bulk/jobs.go (hunk 5 above)
Comment (g8rswimmer) [reply bulk/jobs.go]: This is used for the `all jobs` API and in the `bulk.go` file 
https://developer.salesforce.com/docs/atlas.en-us.api_bulk_v2.meta/api_bulk_v2/get_all_jobs.htm

File: bulk/jobs.go (hunk 6 above)
Comment (g8rswimmer) [reply bulk/jobs.go]: It is the filtering for the get all jobs API

File: bulk/jobs.go (hunk 5 above)
Comment (Guitarbum722) [reply bulk/jobs.go]: Oh I see it now.  Sorry - wasn't able to search the usage of the function call before for some reason.

**Pull request merged**